bridges/snowbridge/docs/v2.md

```diff
@@ -0,0 +1,310 @@
+
+On-chain exchange rate is eliminated. Users pay remote delivery costs in ETH, and this amount is sent with the message as WETH. The delivery fee can be claimed by the relayer on BH.
+
+The user agent applies a similar dry-running process as with [Step 2: User agent estimates fees](https://www.notion.so/Step-2-User-agent-estimates-fees-113296aaabef8159bcd0e6dd2e64c3d0?pvs=21).
+
+The message is converted from $m_0$ to $x_0$ during message submission on BH. Dry-running submission will return $x_0$ to the relayer so that it can verify it is profitable.
+
+## Step 3: AH receives $x_0$ from BH
+
+Submitting the message $m_0$ will cause the following XCM, $x_0$, to be built on BH and dispatched to AH.
+
+```
+WithdrawAsset (KLT, 100)
+ReserveAssetDeposited(WETH, 2)
+PayFees (WETH, 1)
+SetAssetClaimer Bob            // derived from beneficiary
+AliasOrigin Ethereum/Alice     // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+	DepositAsset (KLT, 100) beneficiary=Bob
+```
+
+```
+WithdrawAsset (KLT, 100)
+ReserveAssetDeposited(WETH, 2)
+PayFees (WETH, 1)
+SetAssetClaimer Ethereum/Alice   // derived from beneficiary
+AliasOrigin Ethereum/Alice       // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+		DepositAsset (KLT, 100) beneficiary=Bob
+		Transact Bob.hello()
+```
+
+## Step 4: KILT Receives XCM from AH
+
+The following XCM $x_1$ is received from BH on AH.
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Ethereum/Alice
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+SetAssetClaimer Bob
+DepositAsset (KLT, 100) beneficiary=Bob
+```
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Ethereum/Alice
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+SetAssetClaimer Bob
+DepositAsset (KLT, 100) beneficiary=Bob
+Transact Bob.hello()          // executes with the origin from AH
+```
+
+# Relayer Rewards
+
+The tracking and disbursement of relayer rewards for both directions has been unified. Rewards in WETH are accumulated on BH but must be manually claimed to deposit them into an account.
+
+To claim, call following extrinsic, where $o$ is rewards account (origin), and $w$ is account on AH where the WETH will be paid out.
+
+$$
+\mathrm{claim}(o,w)
+$$
+
+For tax accounting purposes it might be desirable that $o \neq w$.
+
+# Top-Up
+
+Top-up of the relayer reward is viable to implement for either direction as extrinsics on Bridge Hub, respectively Ethereum.
+
+## Origin Preservation
+
+Origins for transact will be preserved by use of the `AliasOrigin` instruction. This instruction will have the following rules that parachain runtimes will need to allow:
+
+1. `AliasOrigin` can behave like `DescendOrigin`. This is allowed because it respects the hierarchy of multi-locations and does not allow jumping up.
```

Comment:
```suggestion
1. `AliasOrigin` can behave like `DescendOrigin`. This is safe because it respects the hierarchy of multi-locations and does not allow jumping up. Meaning no escalation of privileges.
```